eventeditor/actor_json.py

```diff
@@ -0,0 +1,121 @@
+from enum import IntEnum
+import typing
+
+import json
+from pathlib import Path
+import PyQt5.QtWidgets as q
+
+_actor_json_path: typing.Optional[Path] = None
+def set_actor_json_path(p: typing.Optional[str]) -> None:
+    if p:
+        global _actor_json_path
+        _actor_json_path = Path(p)
+
+class EventType(IntEnum):
+    Action = 0
+    Query = 1
+
+def load_actor_json(actor_name: str) -> typing.Dict[str, typing.Any]:
+    if not _actor_json_path:
+        return None
+
+    try:
+        # First look for individual actor file (overrides; for power users)
+        with open(_actor_json_path.parent/f'{actor_name}.json', 'rt') as file:
+            return json.loads(file.read())
+    except:
+        try:
+            # Otherwise look in actor definitions file
+            with open(_actor_json_path, 'rt') as file:
+                return json.loads(file.read())[actor_name]
+        except:
+            return None
+
+def load_event_parameters(actor_name: str, event_name: str, event_type: EventType) -> typing.Dict[str, typing.Any]:
+    try:
+        actor = load_actor_json(actor_name)
+
+        if event_type == EventType.Action:
+            return actor['actions'][event_name]
+        if event_type == EventType.Query:
+            return actor['queries'][event_name]
+        return None
+    except:
+        return None
+
+def load_actions(actor_name: str) -> typing.KeysView[str]:
+    try:
+        return load_actor_json(actor_name)['actions'].keys()
+    except:
+        return None
+
+def load_queries(actor_name: str) -> typing.KeysView[str]:
+    try:
+        return load_actor_json(actor_name)['queries'].keys()
+    except:
+        return None
+
+def export_definitions(flow, widget) -> None:
+    if not flow:
+        q.QMessageBox.information(widget, 'Export actor definition data', 'Open an event flow first')
+        return
+
+    if not _actor_json_path:
+        set_actor_json_path(q.QFileDialog.getSaveFileName(widget, 'Set ',  'actor_definitions', 'JSON (*.json)')[0])
+
+    if not _actor_json_path:
+        return
+
+    try:
+        with open(_actor_json_path, 'rt') as file:
+            definitions = json.loads(file.read())
+    except:
+        definitions = dict()
+    
+    for actor in flow.flowchart.actors:
+        if actor.identifier.name not in definitions:
```

Comment:
```py
actor_def = definitions.get(actor.identifier.name, {})
```
and then use actor_def instead of repeating definitions[actor.identifier.name] all the same

and at the end:
```py
definitions[actor.identifier.name] = actor_def
```